packages/backend-app-api/src/services/implementations/auth/DefaultAuthService.ts

```diff
@@ -82,7 +86,20 @@ export class DefaultAuthService implements AuthService {
 
     const externalResult = await this.externalTokenHandler.verifyToken(token);
     if (externalResult) {
-      return createCredentialsWithServicePrincipal(externalResult.subject);
+      if (
+        externalResult.accessRestrictions &&
+        !externalResult.accessRestrictions.some(
+          r => r.pluginId === this.pluginId,
+        )
+      ) {
+        throw new NotAllowedError('Access restricted');
```

Comment:
Maybe a bit more info here tbh? Thinking about the case where someone provides a token to someone else and they then get this error when making a call. Something like "the provided token is restricted to the '...' plugin(s)". Feels like that handling of user input is relatively safe

Reply:
Sure! Maybe I was overly cautious here. After all, these are tokens that you may hand out to random systems on the internet so if they leak, this lets attackers iterate a bit more easily. But I can add that info